xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.cpp

```diff
@@ -195,6 +206,49 @@ void CRendererBase::Render(CD3DTexture& target, const CRect& sourceRect, const C
       return;
   }
 
+  if (DX::DeviceResources::Get()->Is10BitSwapchain())
+  {
+    if (IsStreamHDR10(buf))
+    {
+      DXGI_HDR_METADATA_HDR10 hdr10 = GetDXGIHDR10MetaData(buf);
+      if (m_isHdrEnabled)
+      {
+        // Only Sets HDR10 metadata if differs from previous
+        if (!CompareDXGIHDR10MetaData(hdr10, m_lastHdr10))
```

Comment:
This method is only used once. Is it better to just inline the compare rather than introducing a new method?

Reply:
Maybe, I prefer it that way because it is clearer and structured... but it can be changed.

Reply:
That's fine, I personally would lean to fewer oneline, single use methods. Have to jump around the code to see what it does, when it's only a single line anyway.

You can leave it, only personal preference really

Reply:
Changed finally to a inline function. I think it improves performance.